gno.land/cmd/gnoland/config_init.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/config"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+)
+
+var errInvalidConfigOutputPath = errors.New("invalid config output path provided")
+
+// newConfigInitCmd creates the config init command
+func newConfigInitCmd(io commands.IO) *commands.Command {
+	cfg := &configCfg{}
+
+	cmd := commands.NewCommand(
+		commands.Metadata{
+			Name:       "init",
+			ShortUsage: "config init [flags]",
+			ShortHelp:  "Initializes the Gno node configuration",
+			LongHelp: "Initializes the Gno node configuration locally with default values, which includes" +
+				" the base and module configurations",
+		},
+		cfg,
+		func(_ context.Context, _ []string) error {
+			return execConfigInit(cfg, io)
+		},
+	)
+
+	return cmd
+}
+
+func execConfigInit(cfg *configCfg, io commands.IO) error {
+	// Check the config output path
+	if cfg.configPath == "" {
+		return errInvalidConfigOutputPath
+	}
```

Comment:
nit: I personally wouldn't write this (if the path is empty, WriteConfigFile will return an error anyway, same as if you try writing it to a dir that doesn't exist) but doesn't matter much

Reply:
I have PTSD from the original config methods that take the liberty of making coffee before creating (and verifying!) the configuration file at a given path

I do this check because I don't want to assume what `WriteConfigFile` does under the hood exactly, or what kind of permissions it has (whether it can create the file if it doesn't exist etc) -- I follow the philosophy of checking user flags top-level, and making the functions themselves oblivious to the context

If it's alright, I'd like to keep this check, primarily for the reasons I've mentioned